odoo/addons/base/models/res_device.py

```diff
@@ -16,62 +17,27 @@
 class ResDeviceLog(models.Model):
     _name = 'res.device.log'
     _description = 'Device Log'
-    _rec_names_search = ['platform', 'browser']
-
-    session_identifier = fields.Char("Session Identifier", required=True, index='btree')
-    platform = fields.Char("Platform")
-    browser = fields.Char("Browser")
-    ip_address = fields.Char("IP Address")
-    country = fields.Char("Country")
-    city = fields.Char("City")
-    device_type = fields.Selection([('computer', 'Computer'), ('mobile', 'Mobile')], "Device Type")
-    user_id = fields.Many2one("res.users", index='btree')
-    first_activity = fields.Datetime("First Activity")
-    last_activity = fields.Datetime("Last Activity", index='btree')
-    revoked = fields.Boolean("Revoked",
+    _rec_names_search = ['ip_address', 'user_agent']
+
+    # Fields that identify a session
+    session_identifier = fields.Char('Session Identifier', required=True, index='btree')
+    user_id = fields.Integer(string='User', required=True, index='btree')
+    # Note:
+    # `user_id` is an Integer to keep a reference to the user even when the user
+    # is deleted. This avoids having a foreign key.
+    # Fields that identify a device for a given session (comes from the HTTP request information
+    ip_address = fields.Char('IP Address', required=True)
+    user_agent = fields.Char('User Agent', required=True)
```

Comment:
You might want to consider truncating the field. Currently, it accepts up to 65536 bytes: `http.client.LineTooLong: got more than 65536 bytes when reading header line`

While regular user agents are around 100-130 bytes, there could be use-case where it is very long, but I don't think it's a big deal if we lose information in such cases.